modules/grpc/otel/otel-protobuf-parser.cpp

```diff
@@ -1393,7 +1402,9 @@ syslogng::grpc::otel::ProtobufParser::process(LogMessage *msg)
 
   gssize len;
   LogMessageValueType log_msg_type;
-  const gchar *type = log_msg_get_value_with_type(msg, logmsg_handle::RAW_TYPE, &len, &log_msg_type);
+
+  /* _parse_metadata() may invalidate the returned char pointer, so a copy is made with std::string */
+  std::string type = log_msg_get_value_with_type(msg, logmsg_handle::RAW_TYPE, &len, &log_msg_type);
```

Comment:
[optional] we could avoid the copy/malloc by mapping the borrowed pointer to an enum and then using that to
determine how we unpack the protobuf.